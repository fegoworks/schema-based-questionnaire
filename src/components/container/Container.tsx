import "./container.scss";
import { FunctionComponent, ReactElement } from "react";

interface ContainerProps {
  children?: ReactElement;
  className?: string;
}
const Container: FunctionComponent<ContainerProps> = ({
  children,
  className,
}) => {
  return (
    <div className={`Container ${className ? className : ""}`}>{children}</div>
  );
};

export default Container;
