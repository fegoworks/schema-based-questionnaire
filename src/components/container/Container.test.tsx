import { render } from "@testing-library/react";
import Container from ".";

const setup = () => {
  function ChildComponent() {
    return <div>Child Element</div>;
  }
  return { ChildComponent };
};

describe("Component - Container", () => {
  it("renders Container component", () => {
    const { ChildComponent } = setup();
    const { container } = render(
      <Container>
        <ChildComponent />
      </Container>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders single react child", () => {
    const { ChildComponent } = setup();
    const { getByText } = render(
      <Container>
        <ChildComponent />
      </Container>
    );
    expect(getByText("Child Element")).toBeInTheDocument();
  });

  it("should render a container with max width 800px", () => {
    const { getByText } = render(
      <Container>
        <>Test</>
      </Container>
    );

    const styles = getComputedStyle(getByText("Test"));
    expect(styles.maxWidth).toBe("800px");
  });

  it("should have a class name of Container", () => {
    const { getByText } = render(
      <Container>
        <>Test</>
      </Container>
    );
    expect(getByText("Test")).toHaveClass("Container");
  });
});
