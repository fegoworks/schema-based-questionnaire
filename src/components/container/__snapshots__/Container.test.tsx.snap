// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component - Container > renders Container component 1`] = `
<div>
  <div
    class="Container "
  >
    <div>
      Child Element
    </div>
  </div>
</div>
`;
